packages/loader/container-loader/src/loader.ts

```diff
@@ -202,10 +203,60 @@ export interface ILoaderProps {
 	readonly protocolHandlerBuilder?: ProtocolHandlerBuilder;
 }
 
+/**
+ * Other props which are used to load a container on top of the basic loader props.
+ * @alpha
+ * @legacy
+ */
+export interface IResolveContainerProps extends ILoaderProps {
+	/**
+	 * The request to resolve the container.
+	 */
+	request: IRequest;
+
+	/**
+	 * Pending local state to be applied to the container.
+	 */
+	pendingLocalState?: string;
+}
+
+/**
+ * Other props which are used to create a detached container on top of the basic loader props.
+ * @alpha
+ * @legacy
+ */
+export interface ICreateDetachedContainerProps extends ILoaderProps {
+	/**
+	 * The code details for the container to be created.
+	 */
+	codeDetails: IFluidCodeDetails;
+
+	canReconnect?: boolean;
+
+	clientDetailsOverride?: IClientDetails;
+}
+
+/**
+ * Other props which are used to rehydrate a detached container on top of the basic loader props.
+ * @alpha
+ * @legacy
+ */
+export interface IRehydrateDetachedContainerProps extends ILoaderProps {
+	/**
+	 * The serialized state returned by calling serialize on another container
+	 */
+	serializedState: string;
+
+	canReconnect?: boolean;
+
+	clientDetailsOverride?: IClientDetails;
```

Comment:
If possible, would be nice to avoid duplicating these properties as compared to ICreateDetachedContainerProps, since I think these should stay in sync.

Reply:
I think it is fine as such because that would require creating another middle interface and we would have same number of interfaces, and the middle interface would cause more confusion.

Reply:
Left another comment in my more recent review, I think these props types need more scrutiny both in contents and how they are organized.